css/css-grid/parsing/grid-auto-flow-valid.html

```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>CSS Grid Layout Module Level 1: parsing grid-auto-flow with valid values</title>
+<link rel="author" title="Eric Willigers" href="mailto:ericwilligers@chromium.org">
+<link rel="help" href="https://drafts.csswg.org/css-grid/#propdef-grid-auto-flow">
+<meta name="assert" content="grid-auto-flow supports the full grammar '[ row | column ] || dense'.">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="support/parsing-testcommon.js"></script>
+</head>
+<body>
+<script>
+test_valid_value("grid-auto-flow", "row");
+test_valid_value("grid-auto-flow", "column");
+test_valid_value("grid-auto-flow", "row dense");
+test_valid_value("grid-auto-flow", "dense column", "column dense");
+
+// Blink/WebKit "dense", Edge/Firefox "row dense"
+test_valid_value("grid-auto-flow", "dense", ["dense", "row dense"]);
```

Comment:
Are Edge/Firefox right here? The spec says:
> Computed value: specified value

Reply:
These tests check the serialization of the specified value, not the computed value.

I suspect the serialization should be "dense":
https://drafts.csswg.org/cssom/#serializing-css-values
> If component values can be omitted or replaced with a shorter representation without changing the meaning of the value, omit/replace them.

Reply:
Then IMHO we shouldn't check both things and some browsers will see a failure and eventually fix the issue (or re-discuss the topic if they don't agree).